app/src/main/java/com/example/calculator/CalculateResultActivity.kt

```diff
@@ -17,9 +21,47 @@ class CalculateResultActivity : AppCompatActivity() {
         // btnPerformCalculation: Dynamically set text on this button on the basis of operation selected
         val btnPerformCalculation = findViewById<Button>(R.id.performCalculation)
 
+        // set the text on button according to operation selected
         btnPerformCalculation.text = getBtnText(selectedOperation)
+
+        // fetch values from EditText and perform calculation on button click
+        btnPerformCalculation.setOnClickListener {
+            val firstNumber = findViewById<EditText>(R.id.firstNumber).text.toString().toInt()
+            val secondNumber = findViewById<EditText>(R.id.secondNumber).text.toString().toInt()
+            findViewById<TextView>(R.id.displayResult).text =
+                getCalculatedResult(selectedOperation, firstNumber, secondNumber)
+        }
     }
 
     // function to set the button text as to which operation has to be performed
-    private fun getBtnText(operation: String) = "PERFORM $operation"
+    private inline fun getBtnText(operation: String) = "PERFORM $operation"
+
+    // function to calculate the final result based on operation selected
+    // takes 3 arguments - operation, firstNumber, secondNumber
```

Comment:
```suggestion
    // takes 3 arguments - operation, firstNumber, secondNumber
    /**
    *
    */
```
Use a comment section instead of commenting each and every line